cosmos/keybase.go

```diff
@@ -81,10 +85,25 @@ func (kb *Keybase) Delete(name, passphrase string, skipPass bool) error {
 }
 
 // Sign is a lock protected version of keys.Sign
+// it also keeps the last private key used in memory for the time set in `keepPrivTime` for performance improvement.
 func (kb *Keybase) Sign(name, passphrase string, msg []byte) ([]byte, crypto.PubKey, error) {
 	kb.mx.Lock()
 	defer kb.mx.Unlock()
-	return kb.kb.Sign(name, passphrase, msg)
+	hash := sha256.Sum256([]byte(name + passphrase))
+	priv := kb.privCache[hash]
```

Comment:
Can we just check the presence of the key instead of the fetched value

```suggestion
	priv, ok := kb.privCache[hash]
	if ok {
```